lredis/cqueues.lua

```diff
@@ -87,7 +100,10 @@ function methods:get_next(new_status, new_error, string_null, array_null)
 	if self.in_transaction or (self.subscribed_to == 0 and self.subscribes_pending == 0) then
 		return nil, "not in subscribe mode"
 	end
-	local resp = protocol.read_response(self.socket, new_status, new_error, string_null, array_null)
+	local null = {}
+	local resp, err_code = protocol.read_response(self.socket, new_status, new_error, null, array_null)
```

Comment:
This changes important semantics: why not pass `string_null`?

Reply:
Because ```protocol.read_response``` now does ```return nil, error``` on error and if ```string_null``` happens to be ```nil``` (which is perfectly valid) then we have no way to differentiate the two.  We need to pass a new temporary ```null``` so that we can do that.

There is no change to semantics here.  If you look further down, after checking for errors there is a comparison to ```null``` and we revert to ```string_null``` in that case.